PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,15 @@
+## Description
+Provide a brief description of the PR's purpose here.
+
+## Todos
+Notable points that this PR has either accomplished or will accomplish.
+* **Developer Interest**
+  - [ ] Feature1
+* **User-Facing for Release Notes**
+  - [ ] Feature2
+
+## Questions
+- [ ] Question1
+
+## Status
+- [ ] Ready to go
```

Comment:
Isn't the purpose of the TODO section to allow for discussion throughout the dev process, not just for summarizing the PR at review-time?  Don't we want that level of involvement?